### ISSUE TITLE
Regarding issues with the WADI dataset

### ISSUE THREAD

**Issue by CCRSRG** [d-ailin/GDN#109]
Hello, may I ask why, after data processing, my training set has 118,800 samples, my test set has 17,280 samples, and there are 1,006 anomalies with an anomaly rate of only 5.82%? There is an error in the official attack time period, which has been identified. Why is the anomaly rate not 5.99%?

**Comment by CCRSRG** [comment]
Could you please send me a copy of the data you processed? My email is beidou920@gmail.com . Thank you very much

**Comment by Yu-20250601** [comment]
I'm having the same issue. Did you ever find a solution or identify the cause?

**Comment by liuyishuai111** [comment]
I'm having the same issue. Did you ever find a solution or identify the cause?

**Comment by CCRSRG** [comment]
> I'm having the same issue. Did you ever find a solution or identify the cause?

I haven't solved this problem

**Comment by CCRSRG** [comment]
> I'm having the same issue. Did you ever find a solution or identify the cause?

I haven't solved this problem

**Comment by liuyishuai111** [comment]
The data you used was from either 2017 or 2019.

**Comment by CCRSRG** [comment]
> The data you used was from either 2017 or 2019.
The 2017 A1

**Comment by liuyishuai111** [comment]
I also used the data from 2017. The abnormality rate of the test set that I calculated was 5.770%. Before downsampling, the original test set you had contained 172,801 data points, among which there were 9,860 anomalies, accounting for approximately 5.705%. How did you calculate it?

**Comment by CCRSRG** [comment]
> I also used the data from 2017. The abnormality rate of the test set that I calculated was 5.770%. Before downsampling, the original test set you had contained 172,801 data points, among which there were 9,860 anomalies, accounting for approximately 5.705%. How did you calculate it?

According to the author's paper requirements, skip the first 2160 lines. Additionally, you need to review the attack time periods in the official document, as there are some errors in it.

**Comment by liuyishuai111** [comment]
I used the author's code for data preprocessing. However, I was unable to obtain the training set of 118,795, the test set of 17,275, and the abnormal rate of 5.99% as the author did. Using the same processing method, my results are: training set of 118,800, test set of 17,280, and anomaly rate of 5.770%.我采用作者对数据预处理的代码，我做不出来作者的训练集118,795个、测试集17,275、异常5.99%。按照相同处理方法我的是训练集118800个、测试集17280个、异常5.770%

**Comment by CCRSRG** [comment]
> I used the author's code for data preprocessing. However, I was unable to obtain the training set of 118,795, the test set of 17,275, and the abnormal rate of 5.99% as the author did. Using the same processing method, my results are: training set of 118,800, test set of 17,280, and anomaly rate of 5.770%.我采用作者对数据预处理的代码，我做不出来作者的训练集118,795个、测试集17,275、异常5.99%。按照相同处理方法我的是训练集118800个、测试集17280个、异常5.770%

我的也是训练集118800个、测试集17280个、但异常率是5.82%   你5.77%应该是有个时段漏了，官方的这个时段弄错了  应该是11号：        ("2017-10-11 10:55:00", "2017-10-11 10:56:27"),  # Attack 9

<img width="1047" height="128" alt="Image" src="https://github.com/user-attachments/assets/0c7e3375-a514-4c25-8667-8f0be96e768f" />

**Comment by liuyishuai111** [comment]
我这个日期这里处理的不对，我进行了修改，把10号修改成了11号，异常率是5.82%，我看了2019年的数据pdf中更新了这出错误。不知道作者是怎么做的5.99%复现不出来，我就按照5.82%来进行做实验了。我看很多异常检测论文没有做数据处理，都是直接用，异常率为5.71%最近几年的论文，但是三四年前的论文都是5.99%。感觉是官方更新了数据导致的

**Comment by liuyishuai111** [comment]
你有研究过SWaT数据集吗，应该用那一年的数据来进行实验。我看有很多年份的数据

**Comment by CCRSRG** [comment]
> 你有研究过SWaT数据集吗，应该用那一年的数据来进行实验。我看有很多年份的数据

我看数据集太大了  几百个G就没弄了

**Comment by Yu-20250601** [comment]
> 你有研究过SWaT数据集吗，应该用那一年的数据来进行实验。我看有很多年份的数据

2015年的吧，数据集有点大，你解决了这个问题吗

**Comment by liuyishuai111** [comment]
我在官方下载了用的2015数据很大，但是下载速度很快，但是我处理的数据异常率和作者原本不一样，训练47,520测试44,991，异常率12.21%

**Comment by Yu-20250601** [comment]
官方给的文件我要下载很久，请问有什么办法解决这个问题吗，以及你找到了数据集异常率不同的原因吗，非常感谢你的回答

**Comment by liuyishuai111** [comment]
下数据的话，我是上午下载的，有时候慢有时快，如果这次太慢了。你就关掉再试一次或者换个时间段就会比较快。异常率不一致我也不清楚是为什么，我看现在的一些论文的异常每个人都不一样相同的数据集，我也不知道为什么可能数据处理方式不一样。但是我是按照作者的方式来处理的也不一样。我找不出来什么问题。如果你有什么方法复现希望分享一下
